core/state/snapshot/snapshot.go

```diff
@@ -528,9 +528,6 @@ func diffToDisk(bottom *diffLayer) *diskLayer {
 		base.genAbort <- abort
 		stats = <-abort
 	}
-	// Put the deletion in the batch writer, flush all updates in the final step.
-	rawdb.DeleteSnapshotRoot(batch)
```

Comment:
This step is unnecessary, the flat state changes will always be flushed atomically.

Reply:
Doesn't hurt to leave in, though. Then we're free to maybe flush the batch if we need to, in some non-mainnet case

Reply:
I would prefer to remove it. As in pathdb, we MUST flush the content in a single batch, including the trie nodes and the flat states, so no reason to keep the tricks.